src/main/java/com/icthh/xm/ms/entity/config/DatabaseConfiguration.java

```diff
@@ -155,7 +158,7 @@ public LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource da
                                                                        CurrentTenantIdentifierResolver currentTenantIdentifierResolverImpl,
                                                                        LocalValidatorFactoryBean localValidatorFactoryBean) {
         Map<String, Object> properties = new HashMap<>();
-        properties.putAll(jpaProperties.getHibernateProperties(dataSource));
+        properties.putAll(jpaProperties.getHibernateProperties(new HibernateSettings()));
```

Comment:
it's not equal replace
it's can make problem when we need customization using spring.jpa.properties.* block
this properties need to be added to this list

Reply:
The only point why this method needs DataSource is to determine `ddl-auto` in case of it's not set explicitly in JpaProperties: set `ddl-auto=create-drop` if datasource is embedded otherwise `none`. Here is discussion (however it's not answer the question): https://stackoverflow.com/questions/49362475/spring-boot-2-unable-to-build-hibernate-sessionfactory

Reply:
see last answer in you link and add 
hibernateProps.putAll(this.jpaProperties.getProperties());

Reply:
done

Reply:
in current implemetanion if we add custom propertie to spring.jpa.properties they all will be added.
So do not need to change something